pinot-integration-tests/src/test/java/com/linkedin/pinot/integration/tests/DeleteAPIHybridClusterIntegrationTest.java

```diff
@@ -0,0 +1,247 @@
+ * two month overlap) then deletes segments different ways to test delete APIs.
+ *
+ */
+public class DeleteAPIHybridClusterIntegrationTest extends HybridClusterIntegrationTest {
+  private static final Logger LOGGER = LoggerFactory.getLogger(RealtimeClusterIntegrationTest.class);
+  private static final String TABLE_NAME = "mytable";
+  private static final String TENANT_NAME = "TestTenant";
+  private static int nRows;
+
+  @BeforeClass
+  public void setUp() throws Exception {
+    super.setUp();
+    org.json.JSONObject response = postQuery("select count(*) from 'mytable'");
+    org.json.JSONArray aggregationResultsArray = response.getJSONArray("aggregationResults");
+    org.json.JSONObject firstAggregationResult = aggregationResultsArray.getJSONObject(0);
+    String pinotValue = firstAggregationResult.getString("value");
+    nRows = Integer.parseInt(pinotValue);
+  }
+
+  private int numRowsReturned() throws Exception{
+    org.json.JSONObject response = postQuery("select count(*) from 'mytable'");
+    org.json.JSONArray aggregationResultsArray = response.getJSONArray("aggregationResults");
+    org.json.JSONObject firstAggregationResult = aggregationResultsArray.getJSONObject(0);
+    String pinotValue = firstAggregationResult.getString("value");
+    return Integer.parseInt(pinotValue);
+  }
+
+  private void waitForTableReady() throws Exception {
+    while (true) {
+      int curOfflineRows = numRowsReturned();
+      if (curOfflineRows != nRows) {
+        Thread.sleep(1000);
+      }
+      else {
+        break;
+      }
+    }
+  }
+
+  // @Test Commenting this out for now because of HLC/LLC
+  public void deleteAllRealtimeSegmentsFromGetAPI() throws Exception{
+    sendGetRequest(ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL).
+        forDeleteAllSegmentsWithTypeWithGetAPI(TABLE_NAME, "realtime"));
+    Thread.sleep(3000);
+
+    String postDeleteSegmentList = sendGetRequest(
+        ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL).forListAllSegments(TABLE_NAME));
+    Assert.assertEquals(getSegmentsFromJson(postDeleteSegmentList, "realtime").size(), 1);
+    Assert.assertNotSame(getSegmentsFromJson(postDeleteSegmentList, "offline").size(), 0);
+
+    repushRealtimeSegments();
+  }
+
+  @Test
+  public void deleteAllOfflineSegmentsFromGetAPI() throws Exception{
+    sendGetRequest(ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL).
+        forDeleteAllSegmentsWithTypeWithGetAPI(TABLE_NAME, "offline"));
+    Thread.sleep(3000);
+
+    String postDeleteSegmentList = sendGetRequest(
+        ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL).forListAllSegments(TABLE_NAME));
+    Assert.assertEquals(getSegmentsFromJson(postDeleteSegmentList, "offline"), null);
+    Assert.assertNotSame(getSegmentsFromJson(postDeleteSegmentList, "realtime").size(), 0);
+
+    repushOfflineSegments();
+  }
+
+  @Override
```

Comment:
add a comment that  we are leaving this out because it is being done in the super class